### PR TITLE
SYN-1945 Support Snowflake TRIM.

### PR DESCRIPTION
**Pull request by zdenal**
# Why
We are getting errors with parsing snowflake `TRIM` expression including removing characters separated by comma.

https://docs.snowflake.com/en/sql-reference/functions/trim

### Example of real failing sql
```sql
WITH sub_coupons AS (
    SELECT
        customer_id,
        subscription_id,
        TRIM(coupons.value:coupon_id, '"') AS coupon_id
    FROM model_staging.subscriptions,
    TABLE(FLATTEN(PARSE_JSON(coupons))) AS coupons
),
```

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -1275,13 +1276,24 @@ impl<'a> Parser<'a> {
                 expr: Box::new(expr),
                 trim_where,
                 trim_what: Some(trim_what),
+                trim_characters: None,
+            })
+        } else if self.consume_token(&Token::Comma) && dialect_of!(self is SnowflakeDialect) {
```
Comment (zdenal) [inline src/parser.rs:1281]: no `FROM` and `Snowflake` and next is `Comma` (by doc Snowflake doesn't support `FROM` syntax) -> `parse characters`

File: tests/sqlparser_common.rs [hunk 2]
```diff
@@ -4967,6 +4967,30 @@ fn parse_trim() {
...
     );
+
+    //keep Snowflake TRIM syntax failing
+    let all_expected_snowflake = TestedDialects {
+        dialects: vec![
+            Box::new(GenericDialect {}),
+            Box::new(PostgreSqlDialect {}),
+            Box::new(MsSqlDialect {}),
+            Box::new(AnsiDialect {}),
+            //Box::new(SnowflakeDialect {}),
```
Comment (zdenal) [inline tests/sqlparser_common.rs:4978]: make sure it is **still** failing on others than snowflake.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: Test Suite
```
Comment (zdenal) [inline .github/workflows/test.yml:1]: Added simple test workflow to see on PRs results of tests.

File: src/ast/mod.rs [hunk 4]
```diff
@@ -994,6 +997,9 @@ impl fmt::Display for Expr {
                 } else {
                     write!(f, "{expr}")?;
                 }
+                if let Some(characters) = trim_characters {
+                    write!(f, ", {}", display_comma_separated(characters))?;
+                }
```
Comment (zdenal) [inline src/ast/mod.rs:1002]: print trim characters after expression if there are

**Review by lustefaniak — APPROVED** [review]

**Pull request merged**